batch/test/test_batch.py

```diff
@@ -565,3 +565,61 @@ def test_verify_no_access_to_metadata_server(self):
         status = j.wait()
         assert status['state'] == 'Success', status
         assert "No credentialed accounts." in j.log()['main'], (j.log()['main'], status)
+
+    def test_verify_can_tcp_to_localhost(self):
+        builder = self.client.create_batch()
+        script = '''
+set -e
+nc -l -p 5000 &
+sleep 5
+echo "hello" | nc -q 1 localhost 5000
+'''.lstrip('\n')
+        j = builder.create_job(os.environ['HAIL_NETCAT_UBUNTU_IMAGE'],
+                               command=['/bin/bash', '-c', script])
+        builder.submit()
+        status = j.wait()
+        assert status['state'] == 'Success', (j.log()['main'], status)
+        assert 'hello\n' == j.log()['main']
+
+    def test_verify_can_tcp_to_127_0_0_1(self):
+        builder = self.client.create_batch()
+        script = '''
+set -e
+nc -l -p 5000 &
+sleep 5
+echo "hello" | nc -q 1 127.0.0.1 5000
+'''.lstrip('\n')
+        j = builder.create_job(os.environ['HAIL_NETCAT_UBUNTU_IMAGE'],
+                               command=['/bin/bash', '-c', script])
+        builder.submit()
+        status = j.wait()
+        assert status['state'] == 'Success', (j.log()['main'], status)
+        assert 'hello\n' == j.log()['main']
+
+    def test_verify_can_tcp_to_self_ip(self):
+        builder = self.client.create_batch()
+        script = '''
+set -e
+nc -l -p 5000 &
+sleep 5
+echo "hello" | nc -q 1 $(hostname -i) 5000
+'''.lstrip('\n')
+        j = builder.create_job(os.environ['HAIL_NETCAT_UBUNTU_IMAGE'],
+                               command=['/bin/sh', '-c', script])
+        builder.submit()
+        status = j.wait()
+        assert status['state'] == 'Success', (j.log()['main'], status)
+        assert 'hello\n' == j.log()['main']
+
+    def test_verify_private_network_is_restricted(self):
+        builder = self.client.create_batch()
```

Comment:
Sorry. Feel free to reject this if you don't like it. Should we have a test that makes sure you can talk to the public internet still?